### PR TITLE
docs: Update Docs for Households

### LINKED ISSUE #4079: [Task] - Developer Migration Guide for V2
### What is the problem this task addresses?

Households introduce some breaking changes outlined in https://github.com/mealie-recipes/mealie/pull/3970. We need a developer migration guide to address these changes and make it easy for existing integrations to migrate to V2.

### Proposed/Possible Solution(s)?

New docs page

### LINKED ISSUE #4080: [Task] - Update Docs for Households
### What is the problem this task addresses?

Households changes a bunch of features with permissions (e.g. `/documentation/getting-started/faq/#how-do-private-groups-and-recipes-work`). Before releasing V2 we need to make sure these are up to date. We probably need to add extra docs for new household features (e.g. how the group/household hierarchy works).

### Proposed/Possible Solution(s)?

Add/update docs once V2 is about ready to go

### PR DESCRIPTION
**Pull request by michael-genson**
## What type of PR is this?

_(REQUIRED)_

- documentation

## What this PR does / why we need it:

_(REQUIRED)_

Adds docs for households and the group/household hierarchy. Also adds a migration guide for developers for V1 -> V2.

## Which issue(s) this PR fixes:

_(REQUIRED)_

Closes https://github.com/mealie-recipes/mealie/issues/4080
Closes https://github.com/mealie-recipes/mealie/issues/4079

### CODE REVIEW

File: docs/docs/documentation/getting-started/features.md [hunk 1]
```diff
@@ -164,6 +163,53 @@ Managing a robust collection of recipes inevitable requires a lot of data. Meali
...
+    shoppinglists(Shopping Lists)
+    integrations("Integrations<br/>(Notifiers, Webhooks)")
+
+    %% Styling
+    classDef primary fill:#E58325,stroke:#B4B7C2,color:#FFFFFF
+    classDef secondary fill:#1976D2,stroke:#B4B7C2,color:#FFFFFF
+
+    classDef default fill:#1F2332,stroke:#E58325,color:#B4B7C2
+    linkStyle default stroke:#B4B7C2
+```
```
Comment (Kuchenpirat) [inline docs/docs/documentation/getting-started/features.md]: Seems like text color is not supported this way in mkdocs. This is  kinda fine for the Mealie, Groups and Households. 
But the contrast with the background color for the other nodes makes them almost illegible.

**Review by Kuchenpirat — CHANGES_REQUESTED** [review]
Looks all good to me besides the flowchart text colors.

File: docs/docs/documentation/getting-started/features.md (hunk 1 above)
Comment (michael-genson) [reply docs/docs/documentation/getting-started/features.md]: Huh, weird. Sure I can just get rid of the styling and let mkdocs do its thing

Comment (Kuchenpirat) [reply docs/docs/documentation/getting-started/features.md]: Yeah, was wondering the same. 
They state say that color does not work for all diagrams but they explicitly mention that it works for flowcharts so i don't know whats up with that 🤷‍♂️

**Review by Kuchenpirat — APPROVED** [review]
All good now 👍

**Pull request merged**